tests/integration/RewardsOnlyGauge/test_rewards_total_transferred_in.py

```diff
@@ -0,0 +1,151 @@
+import brownie
+from brownie import ZERO_ADDRESS, chain
+from brownie.test import strategy
+
+from tests.conftest import approx
+
+
+class StateMachine:
+    """
+    Validate that eventually users claim almost all rewards (except for dust)
+    """
+
+    st_account = strategy("address", length=5)
+    st_value = strategy("uint64", min_value=10 ** 10)
+    st_time = strategy("uint", max_value=365 * 86400)
+    st_reward = strategy("uint64")
+
+    def __init__(
+        self, accounts, rewards_only_gauge, mock_lp_token, reward_contract, coin_reward,
+    ):
+        self.accounts = accounts
+        self.token = mock_lp_token
+        self.liquidity_gauge = rewards_only_gauge
+        self.coin_reward = coin_reward
+
+    def setup(self):
+        self.balances = {i: 0 for i in self.accounts}
+        self.rewards_total = 0
+        self.total_balances = 0
+        self.is_rewards_set = False
+
+    def rule_deposit(self, st_account, st_value):
+        """
+        Make a deposit into the `LiquidityGauge` contract.
+
+        Because of the upper bound of `st_value` relative to the initial account
+        balances, this rule should never fail.
+        """
+        balance = self.token.balanceOf(st_account)
+
+        self.liquidity_gauge.deposit(st_value, {"from": st_account})
+        self.balances[st_account] += st_value
+        self.total_balances += st_value
+
+        assert self.token.balanceOf(st_account) == balance - st_value
+
+    def rule_notify_reward(self, st_reward):
+        """
+        Add rewards only if at least someone has deposits.
+        """
+        if self.total_balances > 0:
+            pre_balance = self.coin_reward.balanceOf(self.liquidity_gauge)
+            if not self.is_rewards_set:
+                coin_rewards = [self.coin_reward] + [ZERO_ADDRESS] * 7
+                self.liquidity_gauge.set_rewards(
+                    ZERO_ADDRESS, b"0x", coin_rewards, {"from": self.accounts[0]}
+                )
```

Comment:
This seems better placed in the `__init__` or prior to even launching the state machine.

Reply:
Ahhh yes, my bad